### PR TITLE
Add cos- longitudinal profile

### PR DESCRIPTION
**Pull request by lucafedeli88**
This PR adds a new longitudinal profile: a truncated cosine, defined essentially as:

```
tn = (t - self.t_peak) / self.tau_fwhm
envelope = (
    np.cos(0.5 * np.pi * tn)
    * (tn > -1)
    * (tn < 1)
    * np.exp(+1.0j * (self.cep_phase + self.omega0 * self.t_peak)
    )
```
~~I called this profile `CosLongitudinalProfile`, but I can rename it `CosineLongitudinalProfile` or `TruncatedCosineLongitudinalProfile` if you prefer~~.

17/12/2024: Upon discussion we agreed to call this profile `CosineLongitudinalProfile`

### CODE REVIEW

File: lasy/profiles/longitudinal/cosine_profile.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        \mathcal{L}(t) = \cos\left({ \frac{\pi}{2} \frac{t-t_{peak}}{\tau_{fwhm}} }\right)
+            \theta\left({ \frac{t-t_{peak}}{\tau_{fwhm}} + 1 }\right)
+            \theta\left({ 1 - \frac{t-t_{peak}}{\tau_{fwhm}}} \right)
+            \exp\left({ + i (\phi_{cep} + \omega_0 t_{peak} ) }\right)
+
+
+    Parameters
+    ----------
+    tau_fwhm : float (in second)
+        The Full-Width-Half-Maximum duration of the intensity distribution of the pulse,
```
Comment (AngelFP) [inline lasy/profiles/longitudinal/cosine_profile.py:28]: the description of the `wavelength` parameter is missing here.

**Review by AngelFP — CHANGES_REQUESTED** [review]
Thanks for this PR! Great to have some added tests. Could you check the small comment below?

File: lasy/profiles/longitudinal/cosine_profile.py (hunk 1 above)
Comment (lucafedeli88) [reply lasy/profiles/longitudinal/cosine_profile.py:28]: You are right! Thanks for having spotted this!

**Review by AngelFP — APPROVED** [review]
Looks great now, thanks!

**Pull request merged**